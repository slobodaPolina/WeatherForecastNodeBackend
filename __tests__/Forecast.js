import React from 'react';
import renderer from 'react-test-renderer';
import Forecast from '../src/components/Forecast';

const city = {
    clouds: {
        all: 40
    },
    cod: 200,
    coord: {
        lat: 51.51,
        lon: -0.13
    },
    dt: 1573292996,
    id: 2643743,
    loaded: true,
    main: {
        humidity: 93,
        pressure: 1008,
        pressureAtm: "1.01",
        pressureMmHg: "756.00",
        temp: "2.48",
        temp_max: 277.59,
        temp_min: 274.26
    },
    name: "London",
    rain: {},
    sys: {
        country: "GB",
        id: 1414,
        sunrise: 1573283248,
        sunset: 1573316474,
        type: 1
    },
    timezone: 0,
    visibility: 5000,
    weather: [{
        description: "haze",
        icon: "50d",
        id: 721,
        main: "Haze"
    }],
    wind: {
        speed: 1.5
    }
};


it('renders forecast correctly', () => {
  const tree = renderer.create(
        <Forecast data={city}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
