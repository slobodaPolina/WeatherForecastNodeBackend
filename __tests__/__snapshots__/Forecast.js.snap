// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders forecast correctly 1`] = `
<div
  className="forecastContainer column"
>
  <div
    className="forecastRow row"
  >
    <div
      className="paramTitle"
    >
      Ветер
    </div>
    <div
      className="paramValue"
    >
      1.5
       meter/sec
    </div>
  </div>
  <div
    className="forecastRow row"
  >
    <div
      className="paramTitle"
    >
      Облачность
    </div>
    <div
      className="paramValue"
    >
      haze
    </div>
  </div>
  <div
    className="forecastRow row"
  >
    <div
      className="paramTitle"
    >
      Давление
    </div>
    <div
      className="paramValue"
    >
      756.00
       mm Hg
    </div>
  </div>
  <div
    className="forecastRow row"
  >
    <div
      className="paramTitle"
    >
      Влажность
    </div>
    <div
      className="paramValue"
    >
      93
      %
    </div>
  </div>
  <div
    className="forecastRow row"
  >
    <div
      className="paramTitle"
    >
      Координаты
    </div>
    <div
      className="paramValue"
    >
      [
      -0.13
      , 
      51.51
      ]
    </div>
  </div>
</div>
`;
